checker.cs

```diff
@@ -1,38 +1,86 @@
 using System;
 using System.Diagnostics;
 
+
 class Checker
+
 {
-    static bool batteryIsOk(float temperature, float soc, float chargeRate) {
-        if(temperature < 0 || temperature > 45) {
+  
+    public static bool temperatureIsOk(float temperature)
+    {
+      if (temperature< 0 || temperature> 45)
```

Comment:
similar code is used for SOC too. there is scope for optimization by reducing duplicate code and increasing code reusability by moving similar code to a utility function. Please check.